### ISSUE TITLE
DR datatype definition incorrect

### ISSUE THREAD

**Issue by carolinerosin** [haffonist/igamt#1163]
The HL7 DR datatype derfinition seems incorrect for version 2.5.1 (I haven't checked the other versions)

IGAMT : 
![dr_igamt](https://cloud.githubusercontent.com/assets/13591511/22437215/2c6ec5ee-e6f5-11e6-9f76-9ad569274fa1.JPG)

HL7 v2.5.1 pdf : 
![dr_hl7](https://cloud.githubusercontent.com/assets/13591511/22437245/4d01ec00-e6f5-11e6-9138-c70b3fbe793e.JPG)

**Comment by haffonist** [comment]
@Snelick,<mailto:robert.snelick@nist.gov> We changed this per Rob request.

________________________________

**Comment by robnist** [comment]
But how was it implemented? Talking to Caroline it seems this was done in a blanket fashion and should not be. Steps:

1. Write code to analyze when a 5 level nesting occurs (save this for later use in the design decision documentation)
2. For ONLY those cases do a data type substitution--that is in some (most?) cases there will not be an issue
3. In the cases it is an issue (nesting > 4) do a data type substitution.  After thinking about it, using a separate DT flavor is better. Caroline used DR_DTM which is a good name.

In summary, identify DR data type where it unravels to a 5th level of nesting and replace that DR with a Data type flavor called DR_DTM in which this flavor is removes the "TS" in between. Caroline can send the flavor.

**Comment by haffonist** [comment]
ok.

________________________________

**Comment by haffonist** [comment]
@robnist: 
If a field (ex. NK1-4) has a datatype (ex. XAD) that has component (ex. XAD.12) with DR datatype then substitute DR with DR_DTM. 
correct ?

**Comment by haffonist** [comment]
@robnist  @carolinerosin I am afraid for this issue, we need to get the list of data types which components has DR that needs to substitute with DR_DTM. An example of one is  XAD. The way the code is written do not gives us any graph information between field,component, datatype that will also us to calculate the depth. We could do so but it will require too much refactoring. 

Can you provide the list of complex datatype that will be affected ?

**Comment by haffonist** [comment]
@robnist anything ?

**Comment by robnist** [comment]
We could provide a list, but this would be very error prone.

I don't see why the following process can't be used:

1. iterate over all base data types for all versions
2. if a component of that data type is DR substitute DR with DR_DTM

I don't think there are any issues of this happening at the sub-component level.
